### PR TITLE
Graph: Move qa into plugin

### PR DESCRIPTION
**Pull request by nik9000**
Move `x-pack/qa/smoke-test-graph-with-security` to
`x-pack/plugin/graph/qa/security` which should make it easier to run all
of the tests with a single command. It also lines up the directories
more closely with newer projects like cross cluster replication.

### CODE REVIEW

File: x-pack/plugin/graph/qa/build.gradle [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import org.elasticsearch.gradle.test.RestIntegTestTask
+
+subprojects {
+    // HACK: please fix this
```
Comment (nik9000) [inline x-pack/plugin/graph/qa/build.gradle]: I'm checking on whether we need this. SQL didn't.

Comment (nik9000) [reply x-pack/plugin/graph/qa/build.gradle]: It wasn't required so I've removed it.

**Comment by nik9000** [comment]
@elasticmachine retest this please

**Review by jasontedor — CHANGES_REQUESTED** [review]
Thanks so much for picking this up @nik9000. I am so happy to see us making these changes, I think it is better productivity-wise and organizationally for individual developers on these sub-plugins.

I don't think this work is enough though. I think that you need to make a change like this to `graph/build.gradle`:

```
diff --git a/x-pack/plugin/graph/build.gradle b/x-pack/plugin/graph/build.gradle
index 069bfa5fbbe..f1bac2e54d4 100644
--- a/x-pack/plugin/graph/build.gradle
+++ b/x-pack/plugin/graph/build.gradle
@@ -15,6 +15,15 @@ dependencies {
     testCompile project(path: xpackModule('core'), configuration: 'testArtifacts')
 }
 
+// add all sub-projects of the qa sub-project
+gradle.projectsEvaluated {
+    project.subprojects
+            .find { it.path == project.path + ":qa" }
+            .subprojects
+            .findAll { it.path.startsWith(project.path + ":qa") }
+            .each { check.dependsOn it.check }
+}
+
 run {
     plugin xpackModule('core')
 }
```

Note that we have similar logic in the `ccr/build.gradle` file. You're welcome to add some logic higher up to add this automatically in a follow-up, but I am pretty sure that something like this is needed for this change.

The difference is that then `:x-pack:plugin:graph:check` will run these tests, whereas with the change you have proposed here it would only run under a full `check`.

Additionally, let us call this `with-security` instead of `security`. Then I think the intent of the tests is clearer.

**Comment by jasontedor** [comment]
One additional point, I think it is good to notify the search team of this change with the `:Search/Graph` label yet I think of this as a `:Core/Build` change too.

**Comment by nik9000** [comment]
> The difference is that then `:x-pack:plugin:graph:check` will run these tests, whereas with the change you have proposed here it would only run under a full `check`.

I talked to @jasontedor about this. I always do things like `./gradlew -p x-pack/plugin/graph check` which works with the change as I proposed it. Apparently it is much more common for folks to do `./gradlew x-pack:plugin:graph:check` which requires the change that @jasontedor asked for. I'm making that change now. And the `security` -> `with-security` change.

File: x-pack/plugin/graph/qa/build.gradle (hunk 1 above)
Comment (nik9000) [reply x-pack/plugin/graph/qa/build.gradle]: I was running the tests wrong. It is required.

**Review by nik9000 — COMMENTED** [review]
@jasontedor I've pushed an update with the changes that you asked for.

**Review by jasontedor — APPROVED** [review]
LGTM.

**Comment by atorok** [comment]
See my comment on #34496 about the Gradle code to pull all the checks together.
+1 on having a central place to deal with this, having to cross reference comments on PRs is a good example of why it would be better not to duplicate small snippets like these throughout the build. Also helps allot with consistency.

**Pull request merged**